staging/src/k8s.io/csi-translation-lib/plugins/in_tree_volume.go

```diff
@@ -165,6 +219,57 @@ func translateTopology(pv *v1.PersistentVolume, topologyKey string) error {
 	return nil
 }
 
+// translateTopologyFromCSIToInTree
+// 1. Remove any existing CSI topologyKey from NodeAffinity
+// 2. Add Kubernetes Topology in the NodeAffinity if it does not exist
+func translateTopologyFromCSIToInTree(pv *v1.PersistentVolume, topologyKey string) error {
+
+	csiTopologyZoneValues := getTopologyValues(pv, topologyKey)
+
+	// Frist remove the CSI topology Key
+	removeTopology(pv, topologyKey)
+
+	// Check if Kubernetes Topology already exist
+	topologyZoneValue := getTopologyValues(pv, v1.LabelTopologyZone)
```

Comment:
would it be worth it to have a wrapper function that returns either ga or beta topology key/value?

Reply:
Done